core/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -476,6 +476,8 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
     }
     val isRightAssociative =
       style.indent.rightAssociativeInfixOperatorsLikeLeftAssociative &&
+        // NOTE. Silly workaround because we call infixSplit from assignment =, see #798
+        formatToken.left.syntax != "=" &&
```

Comment:
So the bug happens when making a plain assignment? So the previous test passed because the "left" was never `=`.